### PR TITLE
publish esphome+ardiuno to platform.io

### PR DESCRIPTION
**Pull request by chrisns**
Publishes the code to platform.io
# Note to maintainers:
Before merging this you'll need to:[create an environment](https://github.com/iMicknl/LoctekMotion_IoT/settings/environments/new) named `platformio_publish`
and create a secret within that environment `PLATFORMIO_TOKEN` secret in github

the contents of that secret you can get from:
```bash
pio account register # if you don't already have an account
# verify the email
pio account login
pio account token
```

After this being merged, I'll update the yaml to reflect pulling in the library from the platform.io registry, and I'll also unpublish where its currently being published to:

 - https://registry.platformio.org/libraries/chrisns/LoctekMotion_IoT_esphome
 - https://registry.platformio.org/libraries/chrisns/LoctekMotion_IoT_arduino

would also be good to publish the raspi to pypi which I'd be happy to work on too if you like

### CODE REVIEW

File: .github/workflows/publish.yaml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+          repository_local: true
+      - uses: marvinpinto/action-automatic-releases@v1.2.1
+        with:
+          repo_token: "${{ secrets.GITHUB_TOKEN }}"
+          automatic_release_tag: ${{ steps.semver.outputs.semantic_version}}
+          prerelease: false
+          title: ${{ steps.semver.outputs.semantic_version}}
+          files: "*"
+
+  arduino:
```
Comment (webknjaz) [inline .github/workflows/publish.yaml:30]: I recommend adding an `environment:` and keeping secrets under that rather than repo-global.

Comment (chrisns) [reply .github/workflows/publish.yaml:30]: updated to use platformio_publish environment

**Comment by iMicknl** [comment]
Sorry for the late reply, to understand, what would be the benefit of this change?

**Comment by chrisns** [comment]
> Sorry for the late reply, to understand, what would be the benefit of this change?

Hey, no worries
Just allows for easier inclusion in config to refer to your c without having to copy and store it locally

**Pull request closed without merging**